app/lib/katello/util/cvecf_migrator.rb

```diff
@@ -0,0 +1,33 @@
+module Katello
+  module Util
+    class CVECFMigrator # used in db/migrate/20220929204746_add_content_view_environment_content_facet.rb
+      def execute!
+        hosts_with_no_cve = []
+        hosts_with_missing_cve = []
+
+        ::Host::Managed.all.each do |host|
+          if ::Katello::ContentView.exists?(id: host.content_facet&.content_view_id) && ::Katello::KTEnvironment.exists?(host.content_facet&.lifecycle_environment_id)
+            cve = ::Katello::ContentViewEnvironment.find_by(content_view_id: host.content_facet&.content_view_id, environment_id: host.content_facet&.lifecycle_environment_id)
+            if cve.blank?
+              hosts_with_no_cve << host
+            end
+          else
+            hosts_with_missing_cve << host if host.content_facet.present?
+          end
+        end
+
+        Rails.logger.info "Found #{hosts_with_no_cve.count} hosts whose lifecycle environment does not have a corresponding ContentViewEnvironment"
+        hosts_with_no_cve.each do |host|
+          Rails.logger.info "Creating CVE for host #{host.id} with content_view_id #{host.content_facet.content_view_id} and lifecycle_environment_id #{host.content_facet.lifecycle_environment_id}"
+          ::Katello::ContentViewEnvironment.create(environment_id: host.content_facet.lifecycle_environment_id, content_view_id: host.content_facet.content_view_id)
+        end
+
+        Rails.logger.info "Found #{hosts_with_missing_cve.count} hosts whose content facet is missing either content_view_id or lifecycle_environment_id"
+        hosts_with_missing_cve.each do |host|
+          Rails.logger.info "Updating host #{host.id} with default content_view_id and lifecycle_environment_id"
+          host.content_facet&.update(content_view_id: ::Katello::ContentView.default.first.id, lifecycle_environment_id: ::Katello::KTEnvironment.library.first.id)
```

Comment:
You need to make sure the CV and Environment belong to the same org as the content_facet